packages/flutter/lib/src/material/navigation_bar.dart

```diff
@@ -0,0 +1,1180 @@
+                ),
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+
+/// Inherited widget for passing data from the [NavigationBar] to the
+/// [NavigationBar.destinations] children widgets.
+///
+/// Useful for building navigation bar destinations using:
+/// `NavigationBarDestinationInfo.of(context)`.
+class NavigationDestinationInfo extends InheritedWidget {
+  /// Adds the information needed to build a navigation bar destination to the
+  /// [child] and descendants.
+  const NavigationDestinationInfo({
+    Key? key,
+    required this.index,
+    required this.totalNumberOfDestinations,
+    required this.selectedAnimation,
+    required this.labelBehavior,
+    required this.onTap,
+    required Widget child,
+  }) : super(key: key, child: child);
+
+  /// Which destination index is this in the navigation bar.
+  ///
+  /// For example:
+  /// ```dart
+  /// NavigationBar(
+  ///   destinations: [
+  ///     NavigationDestination(), // This is destination index 0.
+  ///     NavigationDestination(), // This is destination index 1.
+  ///     NavigationDestination(), // This is destination index 2.
+  ///   ]
+  /// )
+  /// ```
+  ///
+  /// This is required for semantics, so that each destination can have a label
+  /// "Tab 1 of 3", for example.
+  final int index;
+
+  /// How many total destinations are are in this navigation bar.
+  ///
+  /// This is required for semantics, so that each destination can have a label
+  /// "Tab 1 of 4", for example.
+  final int totalNumberOfDestinations;
+
+  /// Indicates whether or not this destination is selected, from 0 (unselected)
+  /// to 1 (selected).
+  final Animation<double> selectedAnimation;
+
+  /// Determines the behavior for how the labels will layout.
+  ///
+  /// Can be used to show all labels (the default), show only the selected
+  /// label, or hide all labels.
+  final NavigationDestinationLabelBehavior labelBehavior;
+
+  /// The callback that should be called when this destination is tapped.
+  ///
+  /// This is computed by calling [NavigationBar.onDestinationSelected]
+  /// with [index] passed in.
+  final VoidCallback onTap;
+
+  /// Returns a non null [NavigationDestinationInfo].
+  ///
+  /// This will return an error if called with no [NavigationDestinationInfo]
+  /// ancestor.
+  ///
+  /// Used by widgets that are implementing a navigation bar destination info to
+  /// get information like the selected animation and destination number.
+  static NavigationDestinationInfo of(BuildContext context) {
+    final NavigationDestinationInfo? result = context
+        .dependOnInheritedWidgetOfExactType<NavigationDestinationInfo>();
+    assert(
+    result != null,
+    'Navigation bar destinations need a NavigationBarDestinationInfo parent, '
+        'usually provided by NavigationBar.',
+    );
+    return result!;
+  }
+
+  @override
+  bool updateShouldNotify(NavigationDestinationInfo oldWidget) =>
```

Comment:
Conventional formatting for expressions like this (e.g. see material/ListTile.dart):
```dart
  bool updateShouldNotify(NavigationDestinationInfo oldWidget) {
    return index != oldWidget.index 
        || totalNumberOfDestinations != oldWidget.totalNumberOfDestinations
        || selectedAnimation != oldWidget.selectedAnimation
        || labelBehavior != oldWidget.labelBehavior
        || onTap != oldWidget.onTap;
  }
```

Reply:
Done